### PR TITLE
[12.0][IMP] sale_brand: assign default brand to Sales Teams

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

**Comment by dreispt** [comment]
It's green now. Can we have a review please?

**Comment by sbejaoui** [comment]
Thank you @dreispt , can you add a unit test please?

**Comment by dreispt** [comment]
I don't have any real business logic in the module, I don't see a relevant test to add.

**Review by sbejaoui — APPROVED** [review]
Ok, it's not a big deal, 

It's true that it's not too much to add a unit test, but it's always safer to have one even for an onchange (prevent breaking override for example)

**Comment by dreispt** [comment]
> it's always safer to have one even for an onchange (prevent breaking override for example)

I'm open to that, but in this case the is a new onchange, so we don't even call super().
Would a test case calling the onchange method do anything useful towards your comment?

**Comment by sbejaoui** [comment]
> > it's always safer to have one even for an onchange (prevent breaking override for example)
> 
> I'm open to that, but in this case the is a new onchange, so we don't even call super().
> Would a test case calling the onchange method do anything useful towards your comment?

I am thinking more about the module evolution (we are just in the beginning) and someone, someday, override it.

But as I said, it's not a big deal, (just trying to keep 100% coverage :) )

File: sale_brand/models/sale_order.py [hunk 1]
```diff
@@ -16,3 +16,7 @@ def _prepare_invoice(self):
                 'brand_id': order.brand_id.id,
             })
         return invoice_vals
+
+    @api.onchange('team_id')
+    def _onchange_team_id(self):
+        self.brand_id = self.team_id.brand_id
```
Comment (sbidoul) [inline sale_brand/models/sale_order.py:22]: Would it make sense to do this only if team_id.brand_id is set?

Comment (dreispt) [reply sale_brand/models/sale_order.py:22]: I have concerns with that behavior:
- I pick a Branded Sales Team, Brand X is set
- The I pick an Unbranded Sales Team, the brand is not cleared, and it stays Brand X
- The Unbranded Team SO will print with the Brand X template, and that will be an defect to the Sales Manager

**Pull request merged**